integration-tests/src/test/java/com/datastax/oss/driver/core/cql/QueryTraceIT.java

```diff
@@ -90,4 +91,23 @@ public void should_fetch_trace_when_tracing_enabled() {
     // Don't want to get too deep into event testing because that could change across versions
     assertThat(queryTrace.getEvents()).isNotEmpty();
   }
+
+  @CassandraRequirement(min = "4.0", description = "Check for added port fields")
+  @Test
+  public void should_fetch_ports() {
+    ExecutionInfo executionInfo =
+        SESSION_RULE
+            .session()
+            .execute(
+                SimpleStatement.builder("SELECT release_version FROM system.local")
+                    .setTracing()
+                    .build())
+            .getExecutionInfo();
+
+    assertThat(executionInfo.getTracingId()).isNotNull();
+
+    QueryTrace queryTrace = executionInfo.getQueryTrace();
+    assertThat(queryTrace.getCoordinatorPort()).isEqualTo(7000);
+    assertThat(queryTrace.getEvents().get(0).getSourcePort()).isEqualTo(7000);
+  }
```

Comment:
I think it would be better to include that in the main test, `should_fetch_trace_when_tracing_enabled`. The condition for ports to be present is:
```java
    boolean expectPorts =
        CCM_RULE.getCassandraVersion().nextStable().compareTo(Version.V4_0_0) >= 0
            && !CCM_RULE.getDseVersion().isPresent();
```
That way we can also assert that they're set to 0 when not expected.